patches/server/0020-Monumenta-Fix-shulker-NoAI-allowing-peeking.patch

```diff
@@ -0,0 +1,38 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Ziggleflig <dmdinnc@gmail.com>
+Date: Sun, 1 May 2022 00:46:53 -0400
+Subject: [PATCH] Monumenta - Fix shulker NoAI allowing peeking
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/monster/Shulker.java b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
+index a9dfe190f46230077e2e1bf9aacbf5375651f216..1077cc08b0f716300d7c278bb345a25d72cb1cbe 100644
+--- a/src/main/java/net/minecraft/world/entity/monster/Shulker.java
++++ b/src/main/java/net/minecraft/world/entity/monster/Shulker.java
+@@ -94,6 +94,7 @@ public class Shulker extends AbstractGolem implements Enemy {
+ 
+     @Override
+     protected void registerGoals() {
++        if (this.isNoAi()) return;
+         this.goalSelector.addGoal(1, new LookAtPlayerGoal(this, Player.class, 8.0F, 0.02F, true));
+         this.goalSelector.addGoal(4, new Shulker.ShulkerAttackGoal());
+         this.goalSelector.addGoal(7, new Shulker.ShulkerPeekGoal());
+@@ -175,7 +176,7 @@ public class Shulker extends AbstractGolem implements Enemy {
+     @Override
+     public void tick() {
+         super.tick();
+-        if (!this.level.isClientSide && !this.isPassenger() && !this.canStayAt(this.blockPosition(), this.getAttachFace())) {
++        if (!this.level.isClientSide && !this.isPassenger() && !this.canStayAt(this.blockPosition(), this.getAttachFace()) && !this.isNoAi()) {
+             this.findNewAttachment();
+         }
+ 
+@@ -235,6 +236,10 @@ public class Shulker extends AbstractGolem implements Enemy {
+     }
+ 
+     private void onPeekAmountChange() {
++        if (this.isNoAi()) {
```

Comment:
* There are plenty of cases where we want shulkers to be open specific amounts without opening or closing further, and there's no return statement to prevent the rest of the function from running.

Reply:
noAI would mean never opens, we can add somethign if you want a specific open value but that's a different setup and likely requires a way to parameterize it into nbteditor. There's no return statement because i want this function to ensure it stays shut, so i set what it read to 0 and let it run.

Reply:
Now that this restores the initial peek amount, I would suggest adding an else statement that updates the initial peek amount should the AI be toggled.